src/main/java/core/basesyntax/operations/ReturnOperation.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.ProductsDto;
+import core.basesyntax.Storege;
+import java.time.LocalDate;
+
+public class ReturnOperation implements FruitOperation {
+    private static final LocalDate TODAY = LocalDate.parse("2020-08-27");
+
+    @Override
+    public void fruitOperation(ProductsDto product) {
+        String fruit = product.getName();
+        int quantity = product.getQuantity();
+        LocalDate expirationDate = product.getDate();
+        if (TODAY.isAfter(expirationDate)) {
+            throw new RuntimeException("Spoiled fruit");
+        }
+        Storege.fruitStorage.merge(fruit, quantity, (oldVal, newVal) -> oldVal + newVal);
+        Storege.expirationDateStorage.put(fruit, expirationDate);
```

Comment:
let's move this logic to `StorageService` class

we will have a `StorageService` field in each of our `FruitOperation` implementation
then, in `fruitOperation()` method (I recommended to rename it to `apply()`) we will simply call `storageService.addToStorage()` or `storageService.removeFromStorage()`